kvstore/pebble/interface.go

```diff
@@ -0,0 +1,28 @@
+package pebble
+
+import (
+	"github.com/pokt-network/smt/kvstore"
+)
+
+// Ensure the PebbleKVStore can be used as an SMT node store
+var _ kvstore.MapStore = (PebbleKVStore)(nil)
+
+// PebbleKVStore is an interface that defines a key-value store
+// that can be used standalone or as the node store for an SMT.
+// This is a superset of the MapStore interface that offers more
+// features and can be used as a standalone key-value store.
+type PebbleKVStore interface {
+	// --- Store methods ---
+	Get(key []byte) ([]byte, error)
+	Set(key, value []byte) error
+	Delete(key []byte) error
+	// --- Lifecycle methods ---
+	Stop() error
+	// --- Accessors ---
+	GetAll(prefixKey []byte, descending bool) (keys, values [][]byte, err error)
+	Exists(key []byte) (bool, error)
+	// Len returns the number of key-value pairs in the store
+	Len() (int, error)
+	// --- Data management ---
+	ClearAll() error
```

Comment:
```suggestion
	// General KVStore Methods - From the SM(S)T MapStore Interface
	kvstore.MapStore
```

This could be done with the `badger` kvstore interface as well and seperate the methods not found in the mapstore interface (like the `Backup`/`Restore` methods).

If `pebble` has any sort of methods like those I would add support for them here as well as this is a standalone package when imported enabling extra features outside of the use of the SM(S)T (even if being used with it) could bring a benefit (even a `// TODO: Expand interface features` will be fine for now.